chore/update text

marcellmueller opened the pull request:
- Page text capitalization updates
- Update all code table descriptions to follow capitalization guidelines
- Change code table imports to `on conflict do nothing`, so they will insert new ones if they are created but not overwrite the current ones. The only new one that we know is coming is a new recreation type, so will have to remember to update the capitalization if that one is surface on the public site.

jimmypalelil commented on frontend/src/data/urls.ts:
```diff
@@ -1,6 +1,5 @@
-    'https://www2.gov.bc.ca/gov/content/sports-culture/recreation/camping-hiking/sites-trails',
-  CONTACT: 'https://www.sitesandtrailsbc.ca/contact-us.aspx',
-} as const;
+export const enum EXTERNAL_LINKS {
```
Love the `enum` approach 👍

jimmypalelil commented on frontend/src/components/rec-resource/RecResourcePage.tsx:
```diff
@@ -136,26 +138,41 @@ const RecResourcePage = () => {
+          title: SectionTitles.MAPS_AND_LOCATION,
         },
-        { href: '#camping', title: 'Camping' },
+        { href: '#camping', title: SectionTitles.CAMPING },
```
any reason why `camping` & `Contact` were not replaced with enums?

jimmypalelil commented on frontend/src/components/rec-resource/RecResourcePage.tsx:
```diff
@@ -238,10 +253,10 @@ const RecResourcePage = () => {
               <SiteDescription
-                description={description}
-                maintenanceCode={maintenance_standard_code}
+                description={description ?? undefined}
```
can we not just make `description` and `maintenanceCode` nullable in SiteDescription component instead of handling it here?

---
[out of scope]
--

even better we could look into configuring openapi cli generator to complete avoid null values and only treat them as optional.

jimmypalelil commented on migrations/rst/sql/V1.0.3__recreation_district.sql:
```diff
@@ -20,16 +33,12 @@ values
-  Mackenzie'),
-('RDQC', 'Queen Charlotte Islands
-  (Haida Gwaii)'),
+('RDPG', 'Prince George-Mackenzie'),
```
i see some districts having space around the `-` and some don't. Not sure if its BC gov mandated convention or just a formatting issue

jimmypalelil commented on migrations/rst/sql/V1.0.3__recreation_district.sql:
```diff
@@ -20,16 +33,12 @@ values
-('RDQC', 'Queen Charlotte Islands
-  (Haida Gwaii)'),
+('RDPG', 'Prince George-Mackenzie'),
+('RDQC', 'Queen Charlotte Islands  ( Haida Gwaii )'),
```
why the extra space before the `(` and around "Haida Gwaii"?

jimmypalelil commented:
Great job on the PR... there were a lot of changes to be made. 👏

I would have spent more time automating it then actually doing it manually lol

marcellmueller commented:
> Great job on the PR... there were a lot of changes to be made. 👏
> 
> I would have spent more time automating it then actually doing it manually lol

Thanks for the good review, I updated it based on your comments in this commit: https://github.com/bcgov/nr-rec-resources/pull/505/commits/cbbf245c2aebcb4094e92498df9716acd92c7f1f

Though I didn't configure the api generator to avoid null values, for some reason I had those two fields as `nullable` instead of just `?` so I updated that and regenerated the api and the component is happy.
